qiskit/circuit/register.py

```diff
@@ -41,10 +41,17 @@ def __init__(self, size, name=None):
 
         # validate (or cast) size
         try:
-            size = int(size)
+            size_int = int(size)
         except Exception:
             raise CircuitError("Register size must be castable to an int (%s '%s' was provided)"
                                % (type(size).__name__, size))
+
+        if not np.isclose(size_int, size):
+            raise CircuitError("Register size must be approximately an integer. " +
```

Comment:
To be more precise we should say it must be positive
```suggestion
            raise CircuitError("Register size must be approximately a positive integer. " +
```

Reply:
Changed.

Though that if-statement is not checking to see if its positive.  By specifying positive here, it may confuse exactly why we are throwing the error.